docs/iaq.rst

```diff
@@ -0,0 +1,35 @@
+JAX Infrequently Asked Questions (IAQ)
+======================================
+
+.. _JAX - Frequently Asked Questions (FAQ): https://jax.readthedocs.io/en/latest/faq.html
+.. _JAX - The Sharp Bits: https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html
+
+This document represents a collection of *infrequently asked questions (IAQ)*. Contributions welcome!
+
+How to apply ``jax.device_put`` on a ``jax.Array` that has different values per host into a 'data' mesh axis?
```

Comment:
what does this mean?

Please don't copy internal code into public docs too (as you did below)

Reply:
This is from https://github.com/google/jax/blob/main/jax/_src/array.py#647

Reply:
Better link (permalink from GitHub) https://github.com/google/jax/blob/d03187e3054fe95c57e0b611ba2e399a9ce168fe/jax/_src/array.py#L647

Reply:
Let's back up a bit.. what does that question mean?

Why do you need a different doc for it? I don't think this question belongs in this doc.